msg/Cargo.toml

```diff
@@ -9,6 +9,10 @@ homepage = { workspace = true }
 license = { workspace = true }
 edition = { workspace = true }
 
+[features]
+default = ["std"]
```

Comment:
This is a general query: few other crates (e.g., `solana-program-error`) have a `std` feature, but it is not enabled by default. Do you think it would be useful to have a common pattern, like `std` is always enabled by default or the other way around?

Reply:
Here since the crate is already published I want to break less stuff. This PR is still a breaking change because some people could already be depending on it with `default-features = false`, but it's at least a smaller change

Reply:
But since it is already "breaking", should we update to how ideally we want to go forward with the `std` feature? I wonder if it will become a bit confusing to have some crates with `std` as default and others without. I feel it would be clearer to have a common pattern.

Reply:
I looked through the repo crates and I think right now the pattern is "enable std by default if it's required for the core functionality of the crate"

Reply:
Sounds good, so in this case the core functionality is the `msg!` macro.